### PR TITLE
Add pretrained models guide

### PR DESCRIPTION
**Pull request by sarahsanders-docker**
### Description
- Created guide on Loading Pretrained Models

### Issue Solved
https://github.com/Codecademy/docs/issues/5869

### Type of Change
- Adding a new entry

### Checklist
- [ ] All writings are my own.
- [ ] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own writing and code.
- [ ] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [ ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [ ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  - Neural Networks
+  - Deep Learning
+  - TensorFlow
+  - PyTorch
+  - Datasets
+  - Scikit-learn
+  - AI
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-example-course'
+  - 'paths/example-path'
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
Title: 'Loading Pre-trained Models'
Description: 'Initializes a model architecture with pre-trained weights learned from a large dataset, enabling transfer learning or direct inference.'
Subjects:
  - 'Machine Learning'
Tags:
  - 'Machine Learning'
  - 'Models'
  - 'Neural Networks'
  - 'Deep Learning'
  - 'TensorFlow'
  - 'PyTorch'
  - 'Datasets'
  - 'Scikit-learn'
  - 'AI'
CatalogContent: 
  - 'learn-python-3'
  - 'paths/computer-science'
```

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+---
+
+**Pretrained models** are a foundational concept in machine learning, allowing developers and researchers to leverage models that have already been trained on extensive datasets. This approach accelerates development and improves accuracy by transferring knowledge from existing solutions to new tasks. In PyTorch, loading pretrained models is straightforward and provides access to many state-of-the-art models via the torchvision library and other community-supported sources.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
**Loading pre-trained models** is a foundational process in machine learning that enables developers and researchers to utilize models already trained on extensive datasets. This practice speeds development and enhances accuracy by reusing learned features from existing solutions for new tasks. In PyTorch, loading pre-trained models is simple and accessible, offering a range of state-of-the-art models through libraries like `torchvision` and other community-contributed sources.
```

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+```python
+from torchvision import models
+
+# Load a pretrained model
+model = models.<model_name>(pretrained=True)
+```
+
+- `<model_name>`: Replace this with the desired model name (e.g., `resnet18`, `vgg16`).
+- `pretrained=True`: Specifies that the model should load pretrained weights.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
- `<model_name>`: The name of the model to load (e.g., `resnet18`, `vgg16`, `mobilenet_v2`, etc.). It must be a valid model name from the `torchvision.models` library.
- `pretrained`: If `True`, the function loads a model initialized with weights pre-trained on the ImageNet dataset. If `False`, it loads a model with random weights.
```

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+## Example
+
+### Using a Pretrained ResNet-18 for Inference
+
+This example demonstrates loading a pretrained ResNet-18 model and using it for image classification.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
This example demonstrates how to load a pre-trained ResNet-18 model and use it for image classification:
```

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    output = model(input_tensor)
+
+# Get the predicted class
+predicted_class = torch.argmax(output[0]).item()
+print(f"Predicted Class: {predicted_class}")
+```
+
+### Fine-Tuning a Pretrained Model
+
+Here is an example of modifying a pretrained ResNet-18 model for a custom classification task:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
The following example shows how to modify a pre-trained ResNet-18 model for a custom classification task:
```

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+```python
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
```py
```

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+# Modify the final fully connected layer to match the number of output classes
+num_classes = 10  # Adjust based on your dataset
+model.fc = nn.Linear(model.fc.in_features, num_classes)
+
+# Define loss and optimizer
+criterion = nn.CrossEntropyLoss()
+optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
+
+# The model is now ready for training on your dataset
```
Comment (mamtawardhani) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md]:
```suggestion
import torch.nn as nn
from torchvision import models

# Load the pretrained ResNet-18 model
model = models.resnet18(pretrained=True)

# Modify the final fully connected layer to match the number of output classes
num_classes = 10  # Replace with the number of classes in your custom dataset
model.fc = nn.Linear(model.fc.in_features, num_classes)

# Define loss and optimizer
criterion = nn.CrossEntropyLoss()              # Loss function for classification
optimizer = torch.optim.Adam(model.parameters(), lr=0.001)  # Optimizer for training

# The model is now ready for fine-tuning on your dataset
```

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @sarahsanders 😄 

The entry looks good for a second round of review 🚀

File: content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+Tags:
+  - 'Machine Learning'
+  - 'Models'
+  - 'Neural Networks'
+  - 'Deep Learning'
+  - 'TensorFlow'
+  - 'PyTorch'
+  - 'Datasets'
+  - 'Scikit-learn'
+  - 'AI'
```
Comment (Radhika-okhade) [inline content/pytorch/concepts/loading-pretrained-models/loading-pretrained-models.md:15]:
```suggestion
  - 'AI'
  - 'Datasets'
  - 'Deep Learning'
  - 'Machine Learning'
  - 'Models'
  - 'Neural Networks'
  - 'PyTorch'
  - 'Scikit-learn'
  - 'TensorFlow'
```

**Review by Radhika-okhade — APPROVED** [review]

**Comment by Radhika-okhade** [comment]
Thank you for contributing to Codecademy Docs @sarahsanders 😄

The entry looks good to be merged. 🚀
